[docs] Explain `IDENTITY` columns in YSQL (#20535)
yugabyte/yugabyte-db: docs/content/preview/api/ysql/the-sql-language/statements/dml_insert.md

Specify a comma-separated list of columns names. If a specified column does not exist, an error is raised. Each of the primary key columns must have a non-null value.

### OVERRIDING SYSTEM VALUE

> When this clause is supplied, any values supplied for identity columns will override the default sequence-generated values.

Reviewer: ```suggestion

When you provide this clause, any values provided for identity columns will override the default sequence-generated values.
```

If an identity column is defined as `GENERATED ALWAYS`, it will raise an error to insert an explicit value (other than `DEFAULT`) without setting either `OVERRIDING SYSTEM VALUE` or `OVERRIDING USER VALUE` clauses. (When an identity column is defined as `GENERATED BY DEFAULT`, `OVERRIDING SYSTEM VALUE` will be the normal behavior and specifying it does nothing, but YugabyteDB allows it as an extension.)

### OVERRIDING USER VALUE

> When this clause is supplied, any values supplied for identity columns will be ignored and the default sequence-generated values will be applied.

Reviewer: ```suggestion

When you provide this clause, any values provided for identity columns is ignored, and the default sequence-generated values will be applied.
```

This clause is helpful when copying values between tables. 

> Writing `INSERT INTO table1 OVERRIDING USER VALUE SELECT * FROM table0` will copy from `table0` all non-identity columns in `table1` while the values of the identity columns in `table1` will be generated by the sequences associated with `table1`.

Reviewer: ```suggestion
Writing `INSERT INTO table1 OVERRIDING USER VALUE SELECT * FROM table0` copies all non-identity columns from `table0`  to `table1`, while the values of the identity columns in `table1` is generated by the sequences associated with `table1`.
```
